rules/TypeDeclaration/PHPStan/ObjectTypeSpecifier.php

```diff
@@ -85,8 +85,9 @@ public function narrowToFullyQualifiedOrAliasedObjectType(
                     return new NonExistingObjectType($className);
                 }
 
-                // only support single @template for now
-                if (count($templateTags) !== 1) {
+                $currentTemplateTag = $templateTags[$className] ?? null;
+
+                if ($currentTemplateTag === null) {
```

Comment:
next reassign `$currentTemplateTag = current($templateTags);` on line 95-96 can be removed to ensure pointed to correct bond template